calculator_test.go

```diff
@@ -5,20 +5,82 @@ import (
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+func TestCalculator(t *testing.T) {
+
+	var tests = []struct {
+		function func(float64, float64) float64
+		want     float64
+		input    []float64
+	}{
+		{function: calculator.Add, input: []float64{2, 2}, want: 4},
```

Comment:
Yes, this is good! We can test all two-operand functions this way, provided their signatures are exactly the same.